src/components/OpenShiftIntercom/OpenShiftIntercomModule.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import { useAtom, useAtomValue, useSetAtom } from 'jotai';
+import { Button } from '@patternfly/react-core/dist/dynamic/components/Button';
+import { Tooltip } from '@patternfly/react-core/dist/dynamic/components/Tooltip';
+import RocketIcon from '@patternfly/react-icons/dist/esm/icons/rocket-icon';
+
+import { openShiftIntercomExpandedAtom, intercomModuleManagerAtom, intercomModuleActionAtom } from '../../state/atoms/openShiftIntercomAtom';
+import './OpenShiftIntercom.scss';
+import { useFlag } from '@unleash/proxy-client-react';
+
+// Intercom positioning constants
+const INTERCOM_CONFIG = {
+  DEFAULT_PADDING: 20,
+  MIN_PADDING: 20,
+} as const;
+
+/**
+ * Calculates safe positioning for Intercom widget relative to a button element.
+ * Ensures the widget doesn't overlap with the button or go outside screen bounds.
+ */
+const calculateSafeIntercomPadding = (buttonRect: DOMRect, windowDimensions: { width: number; height: number }) => {
+  const paddingFromBottom = windowDimensions.height - buttonRect.bottom;
+  const paddingFromRight = windowDimensions.width - buttonRect.right;
+  
+  return {
+    vertical: Math.max(INTERCOM_CONFIG.MIN_PADDING, Math.min(windowDimensions.height, paddingFromBottom)),
+    horizontal: Math.max(INTERCOM_CONFIG.MIN_PADDING, Math.min(windowDimensions.width, paddingFromRight)),
+  };
+};
+
+export type OpenShiftIntercomModuleProps = {
+  className?: string;
+};
+
+const OpenShiftIntercomModule: React.FC<OpenShiftIntercomModuleProps> = ({
+  className,
+}) => {
+  // Initialize atom-based state management
+  const isExpanded = useAtomValue(openShiftIntercomExpandedAtom);
+  const initializeIntercomManager = useSetAtom(intercomModuleManagerAtom);
+  const intercomAction = useSetAtom(intercomModuleActionAtom);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  
```

Comment:
Using jotain atoms introduces another dependency into the chain that is a weak spot. Now that we have the shared store capabilities within Scalprum, it's better to use that. We will only rely on the Scalprum and React which are already shared and versions are strictly enforced across all modules.

We have the docs here: https://github.com/scalprum/scaffolding/blob/main/packages/react-core/docs/shared-stores.md

Or you can look at the starter app, it has an example of implementation within HCC.